### ISSUE TITLE
[3.4] container.dumper.inline_class_loader triggers access level exception

### ISSUE THREAD

**Issue by stefandoorn** [symfony/symfony#25353]
| Q                | A
| ---------------- | -----
| Bug report?      | yes
| Feature request? | no
| BC Break report? | no
| RFC?             | no
| Symfony version  | 3.4.1

I just upgraded from 3.3.13 to 3.4.1, which goes correctly now (had some issues, solved now). I read somewhere that enabling the `container.dumper.inline_class_loader` could dramatically speed up development environments, which I can confirm. Load times for the first page tested on development was 50% faster.

The first request works after clearing cache, but my second request (doesn't matter which page in my system) runs into the following error. Without the parameter enabled, this error does not occur.

```
FatalErrorException:
Compile Error: Access level to Symfony\Cmf\Bundle\MediaBundle\Adapter\ElFinder\PhpcrDriver::$options must be protected (as in class FM\ElFinderPHP\Driver\ElFinderVolumeDriver) or weaker
```

```
class PhpcrDriver extends ElFinderVolumeDriver
```

`$options` in PhpcrDriver is `private`. In ElFinderVolumeDriver it is indeed `protected`. I understand that this is an issue anyway. 

But I'm not sure why it only errors out when this parameter is enabled and not before..

**Comment by nicolas-grekas** [comment]
With the parameter enabled, some classes on the hot path are now always loaded. Previously, this class would have been loaded lazily, and thus the error would been seen when actually using the service.

**Comment by stefandoorn** [comment]
Ok, that sounds logic. I hot fixed it in vendor to temporary bypass it and the next error pops up, also probably from a wrong implementation but it's totally different so wanted to share it with you:

```
PHP Fatal error:  Uncaught Symfony\Component\Debug\Exception\ContextErrorException: Warning: Illegal offset type in isset or empty in /var/www/webshop/vendor/symfony/symfony/src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php:356
Stack trace:
#0 /var/www/webshop/vendor/symfony/symfony/src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php(417): Symfony\Component\DependencyInjection\Dumper\PhpDumper->collectLineage(Object(Symfony\Component\DependencyInjection\Parameter), Array)
#1 /var/www/webshop/vendor/symfony/symfony/src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php(801): Symfony\Component\DependencyInjection\Dumper\PhpDumper->addServiceInclude('lakion_cms.repo...', Object(Symfony\Component\DependencyInjection\Definition), Object(SplObjectStorage))
#2 /var/www/webshop/vendor/symfony/symfony/src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php(856): Symfony\Component\DependencyInjection\Dumper\PhpDumper->addService('lakion_cms.repo...', Object(Symfony\Component\DependencyInjection\Def in /var/www/webshop/vendor/symfony/symfony/src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php on line 356

Fatal error: Uncaught Symfony\Component\Debug\Exception\ContextErrorException: Warning: Illegal offset type in isset or empty in /var/www/webshop/vendor/symfony/symfony/src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php on line 356

Symfony\Component\Debug\Exception\ContextErrorException: Warning: Illegal offset type in isset or empty in /var/www/webshop/vendor/symfony/symfony/src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php on line 356

Call Stack:
   12.4243   61064960   1. Symfony\Component\Debug\ErrorHandler->handleException() /var/www/webshop/vendor/symfony/symfony/src/Symfony/Component/Debug/ErrorHandler.php:0
```

![screen shot 2017-12-06 at 12 54 09](https://user-images.githubusercontent.com/4903082/33660575-943dfd10-da84-11e7-8b63-3ae955cc618a.png)

**Comment by sroze** [comment]
Did you clear your cache after "fixing within the vendors" ? 🤔

**Comment by nicolas-grekas** [comment]
You guys are doing too crazy things :)
Who is using `new Parameter()` to define the class of a service?
Anyway, fixed in #25354

**Comment by stefandoorn** [comment]
Hahaha :) Never knew this vendor bundle was doing that.. :(

@sroze Yes I did, same error.

@nicolas-grekas Change works :-) Thanks!
